src/DataBase/UserDatabase.cs

```diff
@@ -132,5 +129,49 @@ public async Task<UserModel> GetUserAsync(Expression<Func<UserModel, bool>> filt
         {
             return await Database.Users.FirstOrDefaultAsync(filter);
         }
+
+        public async Task ChangeCurrency(int userId, string currencyCode)
+        {
+            var user = await Database.Users.FirstOrDefaultAsync(x => x.Id == userId);
+
+            //user.Currency = currencyCode;
+
+            await Database.SaveChangesAsync();
+        }
+
+        public async Task ChangeCurrency(int userId, string currencyCode, decimal rate)
```

Comment:
This is the one which recalculates as well?

Reply:
It's actually the other one, maybe I should name them differently :D At the moment the one that takes a decimal rate argument is the one that changes the rate, and this one doesn't touch it

Reply:
Atleast it shows to me that this one is the one that takes rate :D so yeah this should recalculate :D

Reply:
Ah I was doing this too late in the night, you're right, the one that is showing rate is this one and it's the one that recalculates :D